### PR TITLE
Create find-replace extension

### PR DESCRIPTION
**Pull request by marekweb**
Demo video: https://www.loom.com/share/58ab1b29a28f407d9ec681fe0fb5cf14?from_recorder=1

### CODE REVIEW

File: src/register-action.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                                    name: currentUser.username,
+                                    email: currentUser.email,
+                                },
+                            },
+                            published: false,
+                        },
+                    })
+            )
+            await sourcegraph.app.activeWindow!.showNotification(
+                `[**Find-replace changes**](${applyURL}) are ready to preview and apply.<br/><br/><small>${diffStat.added} additions, ${diffStat.changed} changes, ${diffStat.deleted} deletions</small>`,
```
Comment (felixfbecker) [inline src/register-action.ts]: **suggestion**: use markdown syntax for paragraphs
```suggestion
                `[**Find-replace changes**](${applyURL}) are ready to preview and apply.\n\n<small>${diffStat.added} additions, ${diffStat.changed} changes, ${diffStat.deleted} deletions</small>`,
```

**Review by tjkandala — APPROVED** [review]
Thanks for adding tests!

File: src/register-action.ts [hunk 2]
```diff
@@ -1,74 +1,82 @@
...
-                                fileFilter,
+                                fileFilter: () => true,
                                 editFile: (path, text) => {
-                                    if (!text.includes(match)) {
+                                    if (!text.includes(findString)) {
+                                        // skip the file by returning null
                                         return null
                                     }
 
                                     percentage += (100 - percentage) / 100
```
Comment (tjkandala) [inline src/register-action.ts]: **suggestion**: Document what's happening with the noop `fileFilter` and `percentage` so we remember what we talked about while pairing when we revisit this

**Comment by marekweb** [comment]
Incorporated latest feedback - thanks!

**Pull request merged**